### DISCUSSION TITLE
The generated image cannot be displayed.

### DISCUSSION THREAD

**Discussion by Lling00** [vladmandic/sdnext#449] [General]
Loading weights: E:\automatic\models\Lora\anotherpureface_v10.safetensors -------------------------- 0.0/9.5 MB -:--:--
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:09<00:00,  2.22it/s]
API error: GET: http://127.0.0.1:7861/file=E:/stable-diffusion-webui/outputs/text/00002-3448286607-An old cabin sits quietly surrounded by trees and the rustling of leaves.  {'error': 'ValueError', 'detail': '', 'body': '', 'errors': 'File cannot be fetched: E:/stable-diffusion-webui/outputs/text/00002-3448286607-An old cabin sits quietly surrounded by trees and the rustling of leaves. . All files must contained within the Gradio python app working directory, or be a temp file created by the Gradio python app.'}
HTTP API: ValueError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ E:\automatic\venv\lib\site-packages\anyio\streams\memory.py:94 in receive                                           │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\anyio\streams\memory.py:89 in receive_nowait                                    │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
WouldBlock

During handling of the above exception, another exception occurred:

┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ E:\automatic\venv\lib\site-packages\starlette\middleware\base.py:78 in call_next                                    │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\anyio\streams\memory.py:114 in receive                                          │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
EndOfStream

During handling of the above exception, another exception occurred:

┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ E:\automatic\modules\middleware.py:68 in exception_handling                                                         │
│                                                                                                                     │
│   67 │   │   try:                                                                                                   │
│ > 68 │   │   │   return await call_next(req)                                                                        │
│   69 │   │   except CancelledError:                                                                                 │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\starlette\middleware\base.py:84 in call_next                                    │
│                                                                                                                     │
│                                              ... 15 frames hidden ...                                               │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\fastapi\routing.py:163 in run_endpoint_function                                 │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\gradio\routes.py:337 in file                                                    │
│                                                                                                                     │
│   336 │   │   │   else:                                                                                             │
│ > 337 │   │   │   │   raise ValueError(                                                                             │
│   338 │   │   │   │   │   f"File cannot be fetched: {path_or_url}. All files must contained wi                      │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
ValueError: File cannot be fetched: E:/stable-diffusion-webui/outputs/text/00002-3448286607-An old cabin sits quietly
surrounded by trees and the rustling of leaves. . All files must contained within the Gradio python app working
directory, or be a temp file created by the Gradio python app.

![image](https://user-images.githubusercontent.com/48402714/234025355-37ab1deb-1f42-4492-a933-79c9318263fd.png)

The generated image is displaying as black, but it is appearing correctly in the saved directory.

**Comment by vladmandic** [comment]
you are running from `E:\automatic`, but your output path is `E:/stable-diffusion-webui`
error message is clear - `All files must contained within the Gradio python app working directory`
this is due to security reasons, otherwise user could fetch any file from your disk.
either change output path or create a symlink to it - in either case, it much be within `E:\automatic`

**Reply by Lling00** [reply]
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:05<00:00,  3.36it/s]
API error: GET: http://127.0.0.1:7860/file=E:/automatic/outputs/text/00001-3448286607-An old cabin sits quietly surrounded by trees and the rustling of leaves.  {'error': 'RuntimeError', 'detail': '', 'body': '', 'errors': 'File at path E:\\automatic\\outputs\\text\\00001-3448286607-An old cabin sits quietly surrounded by trees and the rustling of leaves.  does not exist.'}
HTTP API: RuntimeError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ E:\automatic\venv\lib\site-packages\anyio\streams\memory.py:94 in receive                                           │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\anyio\streams\memory.py:89 in receive_nowait                                    │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
WouldBlock

During handling of the above exception, another exception occurred:

┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ E:\automatic\venv\lib\site-packages\starlette\middleware\base.py:78 in call_next                                    │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\anyio\streams\memory.py:114 in receive                                          │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
EndOfStream

During handling of the above exception, another exception occurred:

┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ E:\automatic\modules\middleware.py:68 in exception_handling                                                         │
│                                                                                                                     │
│   67 │   │   try:                                                                                                   │
│ > 68 │   │   │   return await call_next(req)                                                                        │
│   69 │   │   except CancelledError:                                                                                 │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\starlette\middleware\base.py:84 in call_next                                    │
│                                                                                                                     │
│                                              ... 13 frames hidden ...                                               │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\starlette\routing.py:69 in app                                                  │
│                                                                                                                     │
│ E:\automatic\venv\lib\site-packages\starlette\responses.py:338 in __call__                                          │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
RuntimeError: File at path E:\automatic\outputs\text\00001-3448286607-An old cabin sits quietly surrounded by trees and
the rustling of leaves.  does not exist.

![image](https://user-images.githubusercontent.com/48402714/234040270-0647dd41-f834-425b-991a-b140d63db0ce.png)

Modified, the problem still exists

**Reply by vladmandic** [reply]
did you restart server and generate new image?

**Reply by nnqq2603** [reply]
> you are running from `E:\automatic`, but your output path is `E:/stable-diffusion-webui` error message is clear - `All files must contained within the Gradio python app working directory` this is due to security reasons, otherwise user could fetch any file from your disk. either change output path or create a symlink to it - in either case, it much be within `E:\automatic`

Thanks, this similar issue for me, genetated images failed to display, it generated and saved successfully but can't displayed in UI. Symlinks were already placed for output paths. But still got no image display.

`venv "E:\Stable-diffusion-Vlad\automatic\venv\Scripts\Python.exe"
18:02:52-418013 INFO     Python 3.10.6 on Windows
18:02:52-450030 INFO     Version: 98adfb31 Sun Apr 23 11:50:30 2023 -0400
18:02:52-818019 INFO     Latest published version: 023e13b3714d34b02de5d23fcdb6cb3e5f63e5ab 2023-04-24T23:20:48Z
18:02:52-830033 INFO     Updating Wiki
18:02:55-023167 INFO     nVidia toolkit detected
18:02:56-335192 INFO     Torch 2.0.0+cu118
18:02:56-353199 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
18:02:56-356180 INFO     Torch detected GPU: NVIDIA GeForce GTX 1070 VRAM 8192 Arch (6, 1) Cores 15
18:02:56-358179 INFO     Verifying requirements
18:02:56-405179 INFO     No changes detected: Quick launch active
18:02:56-406181 INFO     Setting environment tuning
18:02:56-408181 INFO     Running extension preloading
18:02:56-410180 INFO     Server arguments: []
No module 'xformers'. Proceeding without it.
Available models: E:\stable-diffusion-webui\models\stable-diffusion 24
ControlNet v1.1.106
ControlNet v1.1.106
Loading theme: black-orange
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Initializing middleware
Loading weights: E:\stable-diffusion-webui\models\stable-diffusion\realisticVisionV20_v20.safetensors ----- 0.0… -:--:…
                                                                                                            GB
Creating model from config: E:\Stable-diffusion-Vlad\automatic\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Applying scaled dot product cross attention optimization
Embeddings loaded: 16-token-negative-deliberate-neg, 3d-female-cyborgs, anders-zorn, Boichi2_style-4000, cancer_style,
cartoona-animals, charturnerv2, chen-1, doose-s-realistic-art-style, dr-strange, durer-style, easynegative,
female-kpop-singer, gta5-artwork, huang-guang-jian, kojima-ayami, midjourney-style, nebula, new-priests,
ng_deepnegative_v1_75t, nixeu, pen-ink-portraits-bennorthen, piotr-jablonski, ralph-mcquarrie, ruan-jia, sakimi-style,
sam-yang, SamDoesArt2, sherhook-painting-v2, sintez-ico, starhavenmachinegods, Style-Empire-neg, Style-Empire,
Style-Moana-neg, Style-Moana, Tomie-Ito, ulzzang-6500-v1.1, ulzzang-6500, vkuoo1, wlop-style, wlop_style (41)
Textual inversion embeddings skipped(13): 21charturnerv2, classipeint, InkPunk768, jah_sidelight, Kipaki-150, mj-gs,
nartfixer, nfixer, nrealfixer, PaintStyle4, retrofutur, rzminjourney, SCG768-Illustrate
Model loaded in 4.0s (load=0.4s create=0.4s apply=0.5s vae=0.5s move=1.2s embeddings=1.1s)
Startup time: 12.3s (torch=1.1s gradio=1.5s libraries=0.6s codeformer=0.2s scripts=1.9s ui=2.5s start=0.3s scripts
app_started_callback=0.1s checkpoint=4.1s)
Progress 1.24it/s ---------------------------------------- 100% 0:00:00 0:00:08
Loading theme: gradio/default
Traceback (most recent call last):
  File "E:\Stable-diffusion-Vlad\automatic\venv\lib\site-packages\gradio\routes.py", line 394, in run_predict
    output = await app.get_blocks().process_api(
  File "E:\Stable-diffusion-Vlad\automatic\venv\lib\site-packages\gradio\blocks.py", line 1075, in process_api
    result = await self.call_function(
  File "E:\Stable-diffusion-Vlad\automatic\venv\lib\site-packages\gradio\blocks.py", line 884, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "E:\Stable-diffusion-Vlad\automatic\venv\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "E:\Stable-diffusion-Vlad\automatic\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
asyncio.exceptions.CancelledError
WebSocket closed (ignore asyncio.exceptions.CancelledError)
Server shutdown
Server restart
Available models: E:\stable-diffusion-webui\models\stable-diffusion 24
ControlNet v1.1.106
Loading theme: gradio/default
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Initializing middleware
Startup time: 2.9s (scripts=0.5s ui=2.2s start=0.1s scripts app_started_callback=0.1s)
Progress 1.7it/s ---------------------------------------- 100% 0:00:00 0:00:05
`

**Reply by nnqq2603** [reply]
> did you restart server and generate new image?

Images don't display on FromText tab, but they do on Image Browser tab. So does it mean anything in my error? Thanks for reading.

**Reply by vladmandic** [reply]
@nnqq2603 only thing that log shows is that you've done restart server from ui and messages are expected.
what happens with a freshly started server (so no restart server from ui)?

**Reply by Lling00** [reply]
> did you restart server and generate new image?

The problem still exists after reboot

**Reply by nnqq2603** [reply]
> @nnqq2603 only thing that log shows is that you've done restart server from ui and messages are expected. what happens with a freshly started server (so no restart server from ui)?

Ah, actually I restart by close/re-run the command line console (server) several times, generate one image, it completed but did not display in the UI. I tried another theme and restart UI, run another generation, image still not display after it finish render. You can see I generate 2 times in the log, it generated image and saved to disk, but did not display in the web page.

I close both and try again now to send you another log.

 `venv "E:\Stable-diffusion-Vlad\automatic\venv\Scripts\Python.exe"
21:30:14-798984 INFO     Python 3.10.6 on Windows
21:30:14-956973 INFO     Version: 98adfb31 Sun Apr 23 11:50:30 2023 -0400
21:30:15-344030 INFO     Latest published version: bb438efed47d5ed064ea5933cccc009d8549d4cb 2023-04-25T13:53:11Z
21:30:15-346006 INFO     Updating Wiki
21:30:18-989250 INFO     nVidia toolkit detected
21:30:22-835304 INFO     Torch 2.0.0+cu118
21:30:22-852305 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
21:30:22-855304 INFO     Torch detected GPU: NVIDIA GeForce GTX 1070 VRAM 8192 Arch (6, 1) Cores 15
21:30:22-857304 INFO     Verifying requirements
21:30:22-920304 INFO     Running setup
21:30:22-922304 INFO     Installing packages
21:30:22-924304 INFO     Installing repositories
21:30:23-128338 INFO     Installing submodules
21:30:26-651331 INFO     Updating submodules
21:30:47-669804 INFO     Extensions enabled: ['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'prompt-bracket-checker', 'ScuNET',
                         'sd-dynamic-thresholding', 'sd-extension-aesthetic-scorer', 'sd-extension-steps-animation',
                         'sd-extension-system-info', 'sd-webui-controlnet', 'sd-webui-model-converter', 'seed_travel',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR']
21:31:08-414851 INFO     Extensions enabled: []
21:31:08-416849 INFO     Setting environment tuning
21:31:08-422850 INFO     Running extension preloading
21:31:08-424851 INFO     Server arguments: []
No module 'xformers'. Proceeding without it.
Available models: E:\stable-diffusion-webui\models\stable-diffusion 24
ControlNet v1.1.107
ControlNet v1.1.107
Loading theme: gradio/default
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Initializing middleware
Loading weights: E:\stable-diffusion-webui\models\stable-diffusion\realisticVisionV20_v20.safetensors ----- 0.0… -:--:…
                                                                                                            GB
Creating model from config: E:\Stable-diffusion-Vlad\automatic\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Applying scaled dot product cross attention optimization
Embeddings loaded: 16-token-negative-deliberate-neg, 3d-female-cyborgs, anders-zorn, Boichi2_style-4000, cancer_style,
cartoona-animals, charturnerv2, chen-1, doose-s-realistic-art-style, dr-strange, durer-style, easynegative,
female-kpop-singer, gta5-artwork, huang-guang-jian, kojima-ayami, midjourney-style, nebula, new-priests,
ng_deepnegative_v1_75t, nixeu, pen-ink-portraits-bennorthen, piotr-jablonski, ralph-mcquarrie, ruan-jia, sakimi-style,
sam-yang, SamDoesArt2, sherhook-painting-v2, sintez-ico, starhavenmachinegods, Style-Empire-neg, Style-Empire,
Style-Moana-neg, Style-Moana, Tomie-Ito, ulzzang-6500-v1.1, ulzzang-6500, vkuoo1, wlop-style, wlop_style (41)
Textual inversion embeddings skipped(13): 21charturnerv2, classipeint, InkPunk768, jah_sidelight, Kipaki-150, mj-gs,
nartfixer, nfixer, nrealfixer, PaintStyle4, retrofutur, rzminjourney, SCG768-Illustrate
Model loaded in 4.7s (load=0.4s create=0.4s apply=0.5s vae=0.6s move=1.2s embeddings=1.6s)
Startup time: 18.2s (torch=1.8s gradio=3.0s libraries=1.8s models=0.5s codeformer=0.3s scripts=2.8s ui=2.8s start=0.3s
scripts app_started_callback=0.1s checkpoint=4.8s)
Progress 1.2it/s ---------------------------------------- 100% 0:00:00 0:00:08
Progress 1.56it/s ---------------------------------------- 100% 0:00:00 0:00:06
`
https://imgur.com/JIclC0O
![image](https://user-images.githubusercontent.com/116576065/234313665-569cfd0f-0622-46cb-be26-156a4bd5eebb.png)

I generated 2 img, no error in console, but images just didn't show up in the UI. Instead, it display like image I uploaded

**Reply by nnqq2603** [reply]
> what happens with a freshly started server (so no restart server from ui)?
I don't see any image newly generated in From Text tab
![image](https://user-images.githubusercontent.com/116576065/234316167-a1b4ea26-8b40-4b6b-98d3-64e6baeab57d.png)

I can find and see images newly generated in Image Browser tab
![image](https://user-images.githubusercontent.com/116576065/234317599-e8122161-2b06-4fe5-8764-1bc3717eb5a7.png)

![image](https://user-images.githubusercontent.com/116576065/234316580-0e464077-aafe-4e24-95a0-848e770faa9c.png)

Fresh session. Finished images was failed to display in UI.

**Reply by Lling00** [reply]
@nnqq2603 How did you set up your theme? I also want to make mine look like yours, it looks more comfortable to look at.

**Reply by nnqq2603** [reply]
> @nnqq2603 How did you set up your theme? I also want to make mine look like yours, it looks more comfortable to look at.

It's a drop-down list you can pick from. Theme now is easy to choose from UI (build-in feature of this fork).
Settings
![image](https://user-images.githubusercontent.com/116576065/234328104-4923d179-bd87-4083-9687-7137eb7bcce4.png)
I switch around themes to see if it does anything, but not, problem come from somewhere else. I got no image display with default _orange black_ one, this one and others. Same issue.
